### PR TITLE
openssl v3.0.2 (redux)

### PR DESCRIPTION
**Pull request by h-vetinari**
Retry #81 

Closes #81 
Closes #79 
Closes #78

### CODE REVIEW

**Comment by h-vetinari** [comment]
Running into problems on OSX: https://github.com/conda/conda-build/issues/4397

**Comment by h-vetinari** [comment]
Thanks a lot @isuruf!

**Comment by h-vetinari** [comment]
Sidenote, I noticed (in #83) that the next rerender will now do
```diff
-c_compiler:
-- clang
-c_compiler_version:
-- '12'
```
for the osx builds. Not sure if you consider this worth fixing, but I wouldn't know how right now (I'm guessing the `compiler` function would need to know how `clang_bootstrap` uses `c_compiler_version`...?)

**Comment by h-vetinari** [comment]
Raised https://github.com/openssl/openssl/issues/17900 for aarch failure

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
 build:
-  number: 2
+  number: 0
 
 requirements:
   build:
-    - {{ compiler('c') }}
+    - {{ compiler('c') }}  # [not osx]
+    # Break circular dependency openssl -> clang -> ld64 -> sigtool -> openssl
+    - {{ compiler('clang_bootstrap') }}  # [osx]
```
Comment (isuruf) [inline recipe/meta.yaml:21]:
```suggestion
    - clang_bootstrap_{{ target_platform }} {{ c_compiler_version }}.*  # [osx]
```

Comment (h-vetinari) [reply recipe/meta.yaml:21]: Thanks!

**Comment by h-vetinari** [comment]
@isuruf, upstream thinks this has something to do with the environment (kernel, QEMU, etc.; worth noting that this is the first PR where we're switching to azure for aarch). If you have any idea what should be providing `io_setup` and how we can add it, that'd be amazing.

```
ALG_PERR: engines/e_afalg.c(252): io_setup error : Function not implemented
```

**Comment by h-vetinari** [comment]
@conda-forge/openssl
It seems we cannot currently build this feedstock for aarch through emulation in QEMU (missing function that should be provided by the kernel). Do we want to just disable the failing test? I don't have other ideas how to fix it.

**Comment by hmaarrfk** [comment]
I'm trying to build natively.

We can likely remove the file:
https://github.com/openssl/openssl/blob/1c0eede9827b0962f1d752fa4ab5d436fa039da4/test/recipes/30-test_afalg.t

**Comment by hmaarrfk** [comment]
Lets see if it works in https://github.com/conda-forge/openssl-feedstock/pull/85

**Comment by hmaarrfk** [comment]
FYI, building natively works. But I'm not interested in making this an other cfep03. skipping the test is likely enough.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by isuruf** [comment]
You can use travis-ci for native aarch64.

**Comment by hmaarrfk** [comment]
Trying here: https://github.com/conda-forge/openssl-feedstock/pull/85#issuecomment-1073372355

**Comment by hmaarrfk** [comment]
Thanks for this

**Comment by h-vetinari** [comment]
@conda-forge/openssl 
This PR and #83 are now ready, and should be merged soon, if at all possible. Thanks for the help @isuruf and @hmaarrfk!

**Comment by h-vetinari** [comment]
Ping @conda-forge/openssl 
(apologies for the higher-than-usual frequency, but leaving CVEs unfixed looks bad on several people's/enterprises' radars, and this PR should be easy to review; same for #83).

**Comment by h-vetinari** [comment]
@conda-forge/core 
Could someone take a look and merge this & #83 if possible (=outstanding CVE fix)? I haven't been successful in getting the attention of the feedstock-team (if desired, I'd join).

**Review by pkgw — APPROVED** [review]
I'm not plugged in enough to the ops to feel comfortable merging in an update of such a fundamental package, but this PR looks good to me given the discussion.

**Comment by xhochy** [comment]
This ended in a near-empty package, see https://github.com/conda-forge/openssl-feedstock/issues/87

**Pull request merged**